### PR TITLE
Remove dependency on Bigchunk, which is a leftover of Cortex and is only used in tests

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

This PR removes the dependency on "Bigchunk", which is a "default" implementation of the `chunk.Data` interface and used only in tests.

**Special notes for your reviewer**:

The PR replaces the usage of this chunk type with the Loki implementation (`chunkenc.MemChunk` wrapped into `chunkenc.Facade`) where possible. In case where it was not possible (tests in `pkg/storage/chunk` package) due to otherwise circular import, it was replaces by the `chunk.dummyChunk` implementation, which is just a no-op that implements `chunk.Data` interface)

The PR also removes the protos for the ingester client (`pkg/ingester/client/ingester.proto`) which were only referenced in the removed `pkg/util/chunkcompat/compat.go` file.

### CODE REVIEW

File: pkg/storage/stores/series/series_store_test.go [hunk 1]
```diff
@@ -173,9 +168,21 @@ func TestChunkStore_LabelValuesForMetricName(t *testing.T) {
 				t.Run(fmt.Sprintf("%s / %s / %s / %s", tc.metricName, tc.labelName, schema, storeCase.name), func(t *testing.T) {
 					t.Log("========= Running labelValues with metricName", tc.metricName, "with labelName", tc.labelName, "with schema", schema)
 					storeCfg := storeCase.configFn()
-					store, _ := newTestChunkStoreConfig(t, schema, storeCfg)
+					store, schemaCfg := newTestChunkStoreConfig(t, schema, storeCfg)
 					defer store.Stop()
 
+					if schema == "v13" {
+						_ = "break"
+					}
```
Comment (chaudum) [inline pkg/storage/stores/series/series_store_test.go:176]: Can be removed. Used for debugging.

File: pkg/storage/chunk/chunk.go [hunk 2]
```diff
@@ -40,11 +40,10 @@ type Chunk struct {
 	logproto.ChunkRef
 
 	Metric labels.Labels `json:"metric"`
+	Data   Data          `json:"-"`
 
-	// We never use Delta encoding (the zero value), so if this entry is
-	// missing, we default to DoubleDelta.
+	// Always use Encoding(129) -> LogChunk
 	Encoding Encoding `json:"encoding"`
-	Data     Data     `json:"-"`
```
Comment (chaudum) [inline pkg/storage/chunk/chunk.go:47]: Undo changes

File: pkg/chunkenc/facade.go [hunk 3]
```diff
@@ -10,17 +10,10 @@ import (
 	"github.com/grafana/loki/pkg/util/filter"
 )
 
-// GzipLogChunk is a cortex encoding type for our chunks.
-// Deprecated: the chunk encoding/compression format is inside the chunk data.
-const GzipLogChunk = chunk.Encoding(128)
```
[the remark this answers is not in the document]
Comment (chaudum) [reply pkg/chunkenc/facade.go:15]: General question: Do we need to support this old chunk format forever? (I'm not opposed)

File: pkg/util/chunkcompat/compat.go [hunk 4]
```diff
@@ -1,53 +0,0 @@
...
-
-	"github.com/prometheus/common/model"
-	"github.com/prometheus/prometheus/model/labels"
-
-	"github.com/grafana/loki/pkg/ingester/client"
-	"github.com/grafana/loki/pkg/storage/chunk"
-)
-
-// FromChunks converts []client.Chunk to []chunk.Chunk.
-func FromChunks(userID string, metric labels.Labels, in []client.Chunk) ([]chunk.Chunk, error) {
```
[the remark this answers is not in the document]
Comment (chaudum) [reply pkg/util/chunkcompat/compat.go:14]: Done in https://github.com/grafana/loki/pull/10354/commits/0fcc6a22aa59db6f2680f4eed6d1b48b519e64d0

**Review by ashwanthgoli — APPROVED** [review]
lgtm

File: pkg/chunkenc/facade.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (slim-bean) [reply pkg/chunkenc/facade.go]: I would say yes, we don't really make guarantees like this but my personal opinion is that I want any version of Loki to always be able to read what was created in the past, this is a nice guarantee to make folks who want to use the product for data stored over many years.

**Review by slim-bean — APPROVED** [review]
LGTM!

File: pkg/chunkenc/facade.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (chaudum) [reply pkg/chunkenc/facade.go]: I've put it back in.

**Review by owen-d — APPROVED** [review]

**Pull request merged**